### PR TITLE
restored the software renderering (non-opengl) waveforms

### PR DESCRIPTION
**Pull request by m0dB**
restored the software renderering (qpainter, non-opengl) waveforms,
moved GL (non-shader) and Qt (polygon-based) to deprecated.

this leaves us with the allshader (OpenGL GLSL based) and software rendering (QPainter based) waveform types.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -689,6 +689,21 @@ endif()
...
+#  src/waveform/renderers/deprecated/glwaveformrenderersimplesignal.cpp
+#  src/waveform/widgets/deprecated/glrgbwaveformwidget.cpp
+#  src/waveform/widgets/deprecated/glsimplewaveformwidget.cpp
+#  src/waveform/widgets/deprecated/glslwaveformwidget.cpp
+#  src/waveform/widgets/deprecated/glwaveformwidget.cpp
+#  src/waveform/widgets/deprecated/qthsvwaveformwidget.cpp
+#  src/waveform/widgets/deprecated/qtrgbwaveformwidget.cpp
+#  src/waveform/widgets/deprecated/qtsimplewaveformwidget.cpp
+#  src/waveform/widgets/deprecated/qtvsynctestwidget.cpp
+#  src/waveform/widgets/deprecated/qtwaveformwidget.cpp
```
Comment (ywwg) [inline CMakeLists.txt:705]: Why this block?

**Review by ywwg — APPROVED** [review]

**Review by Swiftb0y — APPROVED** [review]
LGTM

**Comment by m0dB** [comment]
I will need to discuss with @acolombier on how to move forward. To be honest, I am completely lost what waveform types we want to maintain, but I don't think we are aligned...

**Comment by ywwg** [comment]
let me know if you need any help.  Maybe we should have written a design doc!

**Comment by acolombier** [comment]
I have added [precisions](https://github.com/mixxxdj/mixxx/issues/13226#issuecomment-2124611841) on my understanding of the waveform plan - hopefully it makes up for our lack of design doc.

**Comment by m0dB** [comment]
Replaced by https://github.com/acolombier/mixxx/pull/3

**Pull request closed without merging**